### ISSUE TITLE
Simply Improved Terrain is causing severe generation lag in my pack

### ISSUE THREAD

**Issue by Roadhog360** [KdotJPG/Simply-Improved-Terrain#32]
Apologies for the uninformal initial post. I will update this post later, after doing some proper research, but I wanted to put this here now to make you aware that there is some sort of conflict, I will do more research into it at a later time. [In the meantime, here's a spark profiler report.](https://spark.lucko.me/k6CAeHI2S3)

In my pack, I've narrowed down that the act of simply removing Simply Improved Terrain fixes pretty much any worldgen lag, for some reason when it's installed, the server just cannot keep up, even slow movements quickly overwhelm the server and drag the TPS to its knees.

Please let me know if you'd like me to send you my modpack's files. I will investigate the issue myself sooner or later, as I love SIT's improvements. If you add me on Discord I'll also send the files there if that'd be a preferable place to send files; I have the same username everywhere.

**Comment by KdotJPG** [comment]
Seems it may be [`BiomeFiddleHelper.getFiddledDistance()`](https://github.com/KdotJPG/Simply-Improved-Terrain/blob/1.20/common/src/main/java/jpg/k/simplyimprovedterrain/biome/BiomeFiddleHelper.java#L47) which is called up to 125 times in [`MixinBiomeManager.getBiome()`](https://github.com/KdotJPG/Simply-Improved-Terrain/blob/1.20/common/src/main/java/jpg/k/simplyimprovedterrain/mixin/MixinBiomeManager.java#L34) for every surface block. I'll look into some optimizations for this.

**Comment by Roadhog360** [comment]
Yeah sorry for the lack of information, I will get some time to test this more thoroughly in a few days, but I just wanted to leave this issue as a sort of marker, so you're at least aware that it happens before I get around to testing it.

**Comment by KdotJPG** [comment]
Feel free to try the latest 0.5.6.0-alpha version. It includes an optimization for the BiomeManager mixin.

**Comment by Roadhog360** [comment]
Already nearly a year since I submitted this?? Damn, life really hits hard. Sorry I provided no more information. As a dev, I really should have done better about that. Hopefully I'll be able to look into this soon and also test the new one.
